* Variable chunk_size takes a value of either 512 or 1024, so if it fails with
  the "correct" setting, try the alternate setting as a final attempt to get
  the device working 
diff -r 0ad70995f3b9 libao2/ao_alsa.c
--- a/libao2/ao_alsa.c	Sun Apr 30 16:54:28 2006 +0200
+++ b/libao2/ao_alsa.c	Sun Apr 30 16:54:28 2006 +0200
@@ -615,11 +615,15 @@ static int init(int rate_hz, int channel
 	if ((err = snd_pcm_hw_params_set_period_size_near(alsa_handler, alsa_hwparams, 
 							  &chunk_size, NULL)) < 0)
 	  {
-	    mp_msg(MSGT_AO,MSGL_ERR,MSGTR_AO_ALSA_UnableToSetPeriodSize,
-			    chunk_size, snd_strerror(err));
+	    chunk_size = 512 + 1024 - chunk_size;
+	    if ((err = snd_pcm_hw_params_set_period_size(alsa_handler, alsa_hwparams, 
+	                                                 chunk_size, 0)) < 0) {
+	      mp_msg(MSGT_AO,MSGL_ERR,MSGTR_AO_ALSA_UnableToSetPeriodSize,
+	      chunk_size, snd_strerror(err));
+	    }
 	    return 0;
 	  }
-	else {
+	if (err >= 0) {
 	  mp_msg(MSGT_AO,MSGL_V,"alsa-init: chunksize set to %li\n", chunk_size);
 	}
 	if ((err = snd_pcm_hw_params_set_periods_near(alsa_handler, alsa_hwparams,
